### PR TITLE
solution: 0283-move-zeros updated with java solutions

### PR DESCRIPTION
**Pull request by vigneshshiv**
solution: 0283-move-zeros updated with java solutions

### CODE REVIEW

File: solutions/0200-0299/0283-move-zeroes-easy.md [hunk 1]
```diff
@@ -73,3 +77,58 @@ def moveZeroes(self, nums: List[int]) -> None:
...
+```
+
+</TabItem>
+</Tabs>
+
+## Approach 2: Two pointers Optimal
+
+Iterate through numbers and count the $0$ occurances, while doing so, if any non-zero number present and if the zero's count more than $0$, shift the current number to current `index - 0's count` index. This solves the problem in linear time and optimally.  
+
+For example, the given input is `[1, 3, 0, 0, 12]`, right now 12 has to be placed in first $0$ th position. While iterating, we have count $2$ zero's and we are at the last index. Shift $12$ present in current index $4$ to index $2$, apply `nums[i - zeros_count] = nums[i]` and set $0$ in the current index.
```
Comment (wingkwong) [inline solutions/0200-0299/0283-move-zeroes-easy.md:110]: prefer LaTex over backtick.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: solutions/0200-0299/0283-move-zeroes-easy.md (hunk 1 above)
Comment (vigneshshiv) [reply solutions/0200-0299/0283-move-zeroes-easy.md]: Ok updated.

**Comment by vigneshshiv** [comment]
Updated LaTeX as per suggestion, let me know if any changes needed.

File: solutions/0200-0299/0283-move-zeroes-easy.md [hunk 2]
```diff
@@ -73,3 +77,58 @@ def moveZeroes(self, nums: List[int]) -> None:
...
+    }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Approach 2: Two pointers Optimal
+
+Iterate through numbers and count the $0$ occurances, while doing so, if any non-zero number present and if the zero's count more than $0$, shift the current number to current `index - 0's count` index. This solves the problem in linear time and optimally.
```
Comment (wingkwong) [inline solutions/0200-0299/0283-move-zeroes-easy.md:108]: don't use backtick. Use

```
$index - count_0$
```

File: solutions/0200-0299/0283-move-zeroes-easy.md [hunk 3]
```diff
@@ -73,3 +77,58 @@ def moveZeroes(self, nums: List[int]) -> None:
...
+```
+
+</TabItem>
+</Tabs>
+
+## Approach 2: Two pointers Optimal
+
+Iterate through numbers and count the $0$ occurances, while doing so, if any non-zero number present and if the zero's count more than $0$, shift the current number to current `index - 0's count` index. This solves the problem in linear time and optimally.  
+
+For example, the given input is `[1, 3, 0, 0, 12]`, right now 12 has to be placed in first $0$ th position. While iterating, we have count $2$ zero's and we are at the last index. Shift $12$ present in current index $4$ to index $2$, apply $nums[i - zeroscount] = nums[i]$ and set $0$ in the current index.
```
Comment (wingkwong) [inline solutions/0200-0299/0283-move-zeroes-easy.md:110]: Use

```
$[1, 3, 0, 0, 12]$
```

and 

```
$nums[i - count_0] = nums[i]$
```

File: solutions/0200-0299/0283-move-zeroes-easy.md [hunk 4]
```diff
@@ -73,3 +77,58 @@ def moveZeroes(self, nums: List[int]) -> None:
...
+
+<Tabs>
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@vigneshshiv"/>
+
+```java
+class Solution {
+    public void moveZeroes(int[] nums) {
+        if (nums.length == 1) return;
+        int zeros_count = 0;
```
Comment (wingkwong) [inline solutions/0200-0299/0283-move-zeroes-easy.md:120]: may change to `count0` to match above change

**Review by wingkwong — CHANGES_REQUESTED** [review]
final round

File: solutions/0200-0299/0283-move-zeroes-easy.md (hunk 4 above)
Comment (vigneshshiv) [reply solutions/0200-0299/0283-move-zeroes-easy.md]: Updated

File: solutions/0200-0299/0283-move-zeroes-easy.md (hunk 3 above)
Comment (vigneshshiv) [reply solutions/0200-0299/0283-move-zeroes-easy.md]: Updated

File: solutions/0200-0299/0283-move-zeroes-easy.md (hunk 2 above)
Comment (vigneshshiv) [reply solutions/0200-0299/0283-move-zeroes-easy.md]: Updated

**Pull request merged**